test/E2ETests/E2EApps/E2EApp/E2EApp.csproj

```diff
@@ -1,7 +1,15 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
-  <PropertyGroup>
+  <PropertyGroup Condition="$(FunctionsRuntimeVersion) != '3'">
```

Comment:
For my own interest, I am curious how this variable is present? Is it an environment variable? Explicitly provided in the build command?

Reply:
It's built using [this script](https://github.com/Azure/azure-functions-dotnet-worker/blob/main/setup-e2e-tests.ps1) for both local and pipeline testing. For pipeline testing, this variable is provided in the list of variables you can configure in Azure DevOps (see [here](https://azfunc.visualstudio.com/Azure%20Functions/_build?definitionId=45&_a=summary)).

Reply:
**Quick note:** I do believe this should work if the variable isn't present since this check would catch that and default to v4. For our purposes though (local testing, AzDO) since we use the script i linked above it won't not be present

Reply:
I see, so you are relying on the fact that ADO will be setting this as an environment variable. I would recommend passing this explicitly to improve the local scenario:

`.\tools\devpack.ps1 -E2E -AdditionalPackArgs @("-c","Release")` --> `.\tools\devpack.ps1 -E2E -AdditionalPackArgs @("-c","Release", "-p:FunctionsRuntimeVersion=$FunctionsRuntimeVersion")`

This does work locally because it defaults to v4. However, I don't think it will behave as expected for trying to run v3 locally - it will probably still default to v4 at build.

Reply:
The script has a [mandatory parameter](https://github.com/Azure/azure-functions-dotnet-worker/blob/91a7e52fe1548176ac266873bd64b923f88ef4c6/setup-e2e-tests.ps1#L2) so running locally will pause the script and validate that you pass in 3 or 4. Does that resolve your scenario?

Reply:
The script having that parameter as mandatory or not is not the issue. The issue is that parameter is not forwarded to the msbuild call when running that script. Changing line 100 to what I mentioned above will forward that parameter.